### DISCUSSION TITLE
Ideas for how to rename the contracts

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#232] [Ideas]
During one of our latest meetings in person, we have agreed that it would be clearer to rename the `SablierV2Linear` and the `SablierV2Pro` contracts differently to signal the fact that these contracts enabled closed-ended streaming with capital lockups.

We haven't, however, agreed upon a final new prefix. These are the idea we had:

- `Lockup`
- `CapitalLockup`
- `ClosedEnded`
- `ClosedEndedLockup`
- `CE` (as in, "Closed Ended")
- `Closed`
- `Vesting` (though I know @razgraf doesn't like to reference any particular use case in the contract name)

The `Linear` and the `Pro` suffixes could be optionally shortened to `L` and `P`, respectively.

At the moment, the variant I like the most is `Lockup` because it captures the current logic of creating streams quite accurately. "Lockup" as a term [is used in traditional finance in vesting contexts](https://www.lawinsider.com/clause/lock-up-vesting), as well as in crypto (see [Near's post](https://wiki.near.org/getting-started/near-token/lockups)).

By contrast, the issue with `ClosedEnded` is that it doesn't indicate to users that they have to make a one-time deposit in order for the stream to start. In principle, a stream could be closed-ended while being lockup-free, allowing users to top up.

On the flip side, an open-ended stream must necessarily be lockup-free, because a payment rate per second cannot be calculated from a fixed deposit without a stop time. From this perspective, `ClosedEnded` would be a better prefix, because it would mirror the `OpenEnded` prefix we would presumably use for what we currently call [`SablierV2Payroll`](https://github.com/sablierhq/v2-payroll/).

To sum up, I tend to favor `Lockup`, but if we agree that we will never build a closed-ended contract that is lockup-free and has top ups, then `ClosedEnded` would also be a good fit.

**Comment by maxdesalle** [comment]
Agree that `Lockup` is probably the best option out there. People not familiar with Sablier know what it means, while `ClosedEnded` might not offer the same clarity for people unfamiliar with our protocol.

**Reply by PaulRBerg** [reply]
Thanks, @maxdesalle. `Lockup` has grown on me in the meantime.

**Comment by razgraf** [comment]
I think I like `Lockup` too. It makes the two current contracts into ...
- `SablierV2LinearLockup` / `SablierV2LockupLinear` / `SablierV2LockupL`
- `SablierV2ProLockup` / `SablierV2LockupPro` / `SablierV2LockupP`

Which one should we go with? I have to say the full-names don't look that long (so the shorthand L/P may not be needed after all)

Also, in the context of us possibly launching a "payroll" flavor, the non-lockup Sablier contracts ... what is the antonym we're gonna use for that (the opposite to this "Lockup" variant) ?

**Reply by maxdesalle** [reply]
I think "Unlocked" would make sense in that context, as it's the exact opposite of "LockedUp".

**Reply by PaulRBerg** [reply]
Thanks, @razgraf. I think we should definitely go with `SablierV2LockupLinear` and `SablierV2LockupPro`, since the "Lockup" categorization comes before the linear <> pro distinction (the latter only makes sense with the context of a lockup streaming contract).

I also agree that the full names aren't that long and we can avoid shorthands - I've seen much longer contracts in DeFi/ Etherscan (just think of all the farming forks). The clarity of the full name is well worth it.

> what is the antonym we're gonna use

No need for an antonym! As I said in the OP, if we agree that we're not going to launch a closed-ended contract that is lockup-free and has top ups (which I think we do), we could just launch the payroll contract as `SablierV2OpenEnded`. If anything, I certainly don't like `Unlocked` - it makes one think about rewards and vesting cliffs and whatnot.

Alternatives to `OpenEnded` are:

- `SablierV2Open`
- `SablierV2Infinite`
- `SablierV2Everlasting`
- `SablierV2Perpetual` (though this might give a misleading indication that we are a derivatives protocol, which we don't want)

To be honest, I am on the fence about the name for the payroll contract. But it's not something we need to decide now.

**Comment by andreivladbrg** [comment]
I've been trying to think of a better name for the last few days and I couldn't find a better one.

The one I find to be the best is `Lockup`. 

I think we all are on the same idea.

**Comment by PaulRBerg** [comment]
Locking, we ended up going with `LockupLinear` and `LockupPro`.
